mods/default/functions.lua

```diff
@@ -366,8 +364,7 @@ minetest.register_abm({
 		end
 
 		-- Look for likely neighbors.
-		local p2 = minetest.find_node_near(pos, 1, {"default:dirt_with_grass",
-				"default:dirt_with_dry_grass", "default:dirt_with_snow"})
+		local p2 = minetest.find_node_near(pos, 1, "group:spreads_on_dirt")
```

Comment:
Hm 'grassy_dirt_spread' .. 'spreading_grassy_dirt'
I'm trying to be more generalised as the surface may not be grassy.
'spreading_dirt_node' ..